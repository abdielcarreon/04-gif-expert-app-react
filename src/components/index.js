

export * from './AddCategory';
export * from './AddCategory';
export * from './GiftGrid';