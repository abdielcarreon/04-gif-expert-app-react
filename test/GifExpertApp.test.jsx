import { render, screen, fireEvent } from "@testing-library/react"
import { GiftExpertApp } from "../src/GiftExpertApp"


describe('Testing on <GiftExpertApp/>', () => {
// Validations: Writing in the input value,
    // Actions in the form, taking snapshot, what happen to the form change, what happened if gives the same category again, what happened if I gives another different category and what happen in the HTML regard of that, etc.
  test('should to match snapshot', () => {
    render(<GiftExpertApp /> )
    const { container } = render(<GiftExpertApp /> )
    
    expect( container ).toMatchSnapshot();
  })
  
    
  test('should write in the input and set the value through the submit', () => {


    render(<GiftExpertApp /> )
    
    const inputValue = 'Saitama';
    const input = screen.getByRole('textbox');
    const form =screen.getByRole('form');
    
    fireEvent.input( input, { target: { value: inputValue }});
    fireEvent.submit(form)
    screen.debug();
    
    
  });
  
})
