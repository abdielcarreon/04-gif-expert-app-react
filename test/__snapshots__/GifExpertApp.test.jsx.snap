// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on <GiftExpertApp/> should to match snapshot 1`] = `
<div>
  <h1>
    GiftExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
